provider/server/machine_volume_detach_test.go

```diff
@@ -0,0 +1,164 @@
+							},
+							Device: "oda",
+						},
+						{
+							Name: "disk-2",
+							EmptyDisk: &iri.EmptyDisk{
+								SizeBytes: emptyDiskSize,
+							},
+							Device: "odb",
+						},
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createResp).NotTo(BeNil())
+
+		DeferCleanup(func(ctx SpecContext) {
+			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
+				Expect(err).ShouldNot(HaveOccurred())
+				_, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+				return libvirt.IsNotFound(err)
+			}).Should(BeTrue())
+		})
+
+		By("ensuring domain and domain XML is created for machine")
+		var domain libvirt.Domain
+		Eventually(func() error {
+			domain, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+			return err
+		}).WithTimeout(5 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+		domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(domainXMLData).NotTo(BeEmpty())
+
+		By("ensuring domain for machine is in running state")
+		Eventually(func() libvirt.DomainState {
+			domainState, _, err := libvirtConn.DomainGetState(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			return libvirt.DomainState(domainState)
+		}).Should(Equal(libvirt.DomainRunning))
+
+		By("ensuring machine is in running state and other status fields have been updated")
+		Eventually(func() *iri.MachineStatus {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).NotTo(BeEmpty())
+			Expect(len(listResp.Machines)).Should(Equal(1))
+			return listResp.Machines[0].Status
+		}).Should(SatisfyAll(
+			HaveField("Volumes", ContainElements(
+				&iri.VolumeStatus{
+					Name:   "disk-1",
+					Handle: "libvirt-provider.ironcore.dev/empty-disk/disk-1",
+					State:  iri.VolumeState_VOLUME_ATTACHED,
+				},
+				&iri.VolumeStatus{
+					Name:   "disk-2",
+					Handle: "libvirt-provider.ironcore.dev/empty-disk/disk-2",
+					State:  iri.VolumeState_VOLUME_ATTACHED,
+				})),
+			HaveField("State", Equal(iri.MachineState_MACHINE_RUNNING)),
+		))
+
+		By("ensuring both the empty disks are attached to a machine domain")
+		var disks []libvirtxml.DomainDisk
+		Eventually(func() int {
+			domainXMLData, err := libvirtConn.DomainGetXMLDesc(domain, 0)
+			Expect(err).NotTo(HaveOccurred())
+			domainXML := &libvirtxml.Domain{}
+			err = domainXML.Unmarshal(domainXMLData)
+			Expect(err).NotTo(HaveOccurred())
+			disks = domainXML.Devices.Disks
+			return len(disks)
+		}).Should(Equal(3))
+		Expect(disks[0].Serial).To(HavePrefix("oda"))
+		Expect(disks[1].Serial).To(HavePrefix("odb"))
+
+		time.Sleep(20 * time.Second)
```

Comment:
pls add comment why is sleep here. thx

Reply:
Added comment  "wait for boot image to be available before detaching volume"